cmd2/argparse_completer.py

```diff
@@ -428,159 +217,177 @@ def process_action_nargs(action: argparse.Action, arg_state: AutoCompleter._Argu
-                if remainder['arg'] is not None:
-                    if remainder['action'] == pos_action:
+            # Are we consuming flag arguments?
+            if flag_arg_state is not None and flag_arg_state.needed:
+                consume_flag_argument()
+            else:
+                if not skip_remaining_flags:
+                    # Special case when each of the following is true:
+                    #   - We're not in the middle of consuming flag arguments
+                    #   - The current positional argument count has hit the max count
+                    #   - The next positional argument is a REMAINDER argument
+                    # Argparse will now treat all future tokens as arguments to the positional including tokens that
+                    # look like flags so the completer should skip any flag related processing once this happens
+                    if pos_arg_state is not None and pos_arg_state.count >= pos_arg_state.max and \
+                            next_pos_arg_index < len(self._positional_actions) and \
+                            self._positional_actions[next_pos_arg_index].nargs == argparse.REMAINDER:
+                        skip_remaining_flags = True
+
+                # At this point we're no longer consuming flag arguments. Is the current argument a potential flag?
+                if is_potential_flag(token, self._parser) and not skip_remaining_flags:
+                    # Reset flag arg state but not positional tracking because flags can be
+                    # interspersed anywhere between positionals
+                    flag_arg_state = None
+                    action = None
+
+                    # does the token fully match a known flag?
+                    if token in self._flag_to_action:
+                        action = self._flag_to_action[token]
+                    elif hasattr(self._parser, 'allow_abbrev') and self._parser.allow_abbrev:
+                        candidates_flags = [flag for flag in self._flag_to_action if flag.startswith(token)]
+                        if len(candidates_flags) == 1:
+                            action = self._flag_to_action[candidates_flags[0]]
+
+                    if action is not None:
+                        flag_arg_state = AutoCompleter._ArgumentState(action)
+
+                        # It's possible we already have consumed values for this flag if it was used earlier
+                        # in the command line. Reset them now for this use of the flag.
+                        consumed_arg_values[flag_arg_state.action.dest] = []
+
+                        # Keep track of what flags have already been used
+                        # Flags with action set to append, append_const, and count can be reused
+                        if not is_last_token and \
+                                not isinstance(flag_arg_state.action, argparse._AppendAction) and \
```

Comment:
Just a thought, but this check could be line reduced to
`not isinstance(flag_arg_state.action, (argparse._AppendAction, argparse._AppendConstAction, argparse._CountAction))`

since it's equivalent to `not (a or b or c)` -> `not a and not b and not c`

Reply:
Good suggestion. I didn't realize isinstance() could take a tuple of types.